# ðŸ§¬ beginning-metagenomics

This repo would serve as beginner's path for metagenomics, while myself a beginner trying to figure out the path and constantly pushing resources and pipline materials.

---

## ðŸ”¬ Amplicon Sequencing

Amplicon sequencing is a method of targeted next generation sequencing (NGS) allowing researchers to analyze genetic variations in specific genomic regions using polymerase chain reaction (PCR) primers designed to amplify a region (regions) of interest in a genome.

The 16S ribosomal rRNA gene in prokaryotes and 18S rRNA gene in eukaryotes is considered as molecular marker for distinguishing.

16S rRNA gene contains 10 highly conserved regions and 9 regions that are variable. In 16S rRNA amplicon sequencing of bacteria, the highly conserved region serves as primer while the variable regions helps in identifying bacterial taxas.

The amplified 16S rRNA gene can then be sequenced on one of numerous platforms including:

- 454-pyrosequencing  
- Illumina Hiseq and Miseq  
- Ion Torrent

---

## ðŸ§ª QIIME2 - Amplicon Distribution

### ðŸ“¦ Artifacts and Visualization

All files generated by QIIME 2 are either `.qza` or `.qzv` files, and these are simply zip files that store your data alongside some QIIME 2-specific metadata.

- `.qza` stands for **QIIME zipped artifacts**, which are intermediary files in an analysis workflow.  
- `.qzv` stands for **QIIME zipped visualization**, which helps in providing interactive visualizations. These files can be viewed in [QIIME2 View](https://view.qiime2.org/)

QIIME 2 artifacts are assigned exactly one artifact class, which indicates the semantics of the data (its semantic type) and the file format that is used to store it inside of the `.qza` file.

---

### ðŸ“œ Data Provenance

QIIME 2 was designed to automatically document analysis workflows for users, ensuring that their bioinformatics work is reproducible.

To achieve this, each QIIME 2 command is recorded when it is run, and that information is stored in all Artifacts and Visualizations that are created.

You can view the data provenance at [QIIME2 View](https://view.qiime2.org/) by simply uploading the `.qza` or `.qzv` file.

---

### ðŸ”Œ Plugins

Plugins define actions, which are the individual commands that youâ€™ll run in an analysis workflow. Three types of actions can be defined by plugins:

- **Methods**: take one or more artifact to generate artifacts.  
- **Visualizers**: take artifact(s) to produce `.qzv` files.  
- **Pipelines**: take artifacts and parameters as input, and produce `.qza` and `.qzv` files.

---

### ðŸ§­ Workflow

![workflow](concept-overview.webp)

Steps of the workflow:

- Importing the raw sequence â€” creation of first `.qza` file from FASTQ file  
- Demultiplexing  
- Denoising and Clustering  
- Taxonomic Classification  

---

### ðŸ”€ Demultiplexing

**Multiplexing** refers to sequencing different samples together, with barcodes attached on one or both ends of each sequence.  
**Demultiplexing** is the process of detecting the barcode sequences and mapping them back to the sample they belong.

These barcode info will be available in the sample metadata file, which can be given as input during demultiplexing.

---

### ðŸ§¹ Denoising and Clustering

Denoising removes low-quality reads. Denoising methods available are **DADA2** and **Deblur**, which produce **Amplicon Sequence Variants (ASVs)**.

Clustering helps in removing chimers or similar sequences and replicates them into one representative â€” known as **OTU (Operational Taxonomic Unit) picking**.

The outputs are:
- A `FeatureTable[Frequency]` artifact  
- A `FeatureData[Sequence]` (representative sequences) artifact  

---

## ðŸ“– References

1. R. Lanza & A. Atala (Eds.), *Molecular Biology*, Academic Press, 2020. [Link](https://www.sciencedirect.com/book/9780128132883/molecular-biology)  
2. Paragon Genomics. *What is Amplicon Sequencing?* [Link](https://www.paragongenomics.com/targeted-sequencing/amplicon-sequencing)  
3. Ellermann, M., Carr, J.S., Fodor, A.A., Arthur, J.C., & Carroll, I.M. (2019). *Characterizing and Functionally Defining the Gut Microbiota: Methodology and Implications*. In: *The Gut-Brain Axis* (Chapter 2), Academic Press.  
4. QIIME2 Docs. *Getting Started with QIIME 2 Amplicon Sequencing.* [Link](https://amplicon-docs.qiime2.org/en/latest/explanations/getting-started.html)  

---
